From 4006f03c918b8fb61df9692faad886c858a3086b Mon Sep 17 00:00:00 2001
From: Suman Mukherjee <sumam@codeaurora.org>
Date: Thu, 11 Jan 2018 14:28:21 +0530
Subject: [PATCH 2/2] SnapdragonCamera: Change static nature for persist flag

Issue: Due to static declaration of flip flags, flip effect
is not getting reflected until the application will get stopped.

Change-Id: If47653a365c840ce262fe9c270472cf41e7dbdba
Fix: In place of static, use flip flags as normal variable.
---
 src/com/android/camera/PhotoModule.java      |  8 +++++---
 src/com/android/camera/VideoModule.java      |  9 ++++++---
 src/com/android/camera/util/PersistUtil.java | 12 ++++++------
 3 files changed, 17 insertions(+), 12 deletions(-)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index 8c6a86f83..afc599e9b 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -4057,9 +4057,11 @@ public class PhotoModule
         //value: 1 - FLIP_MODE_H
         //value: 2 - FLIP_MODE_V
         //value: 3 - FLIP_MODE_VH
-        int preview_flip_value = PersistUtil.getPreviewFlip();
-        int video_flip_value = PersistUtil.getVideoFlip();
-        int picture_flip_value = PersistUtil.getPictureFlip();
+        PersistUtil myUtil     = new PersistUtil();
+        int preview_flip_value = myUtil.getPreviewFlip();
+        int video_flip_value   = myUtil.getVideoFlip();
+        int picture_flip_value = myUtil.getPictureFlip();
+
         int rotation = CameraUtil.getJpegRotation(mCameraId, mOrientation);
         mParameters.setRotation(rotation);
         if (rotation == 90 || rotation == 270) {
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 28766f4aa..14732483c 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -2437,9 +2437,12 @@ public class VideoModule implements CameraModule,
         //value: 1 - FLIP_MODE_H
         //value: 2 - FLIP_MODE_V
         //value: 3 - FLIP_MODE_VH
-        int preview_flip_value = PersistUtil.getPreviewFlip();
-        int video_flip_value = PersistUtil.getVideoFlip();
-        int picture_flip_value = PersistUtil.getPictureFlip();
+
+        PersistUtil myUtil     = new PersistUtil();
+        int preview_flip_value = myUtil.getPreviewFlip();
+        int video_flip_value   = myUtil.getVideoFlip();
+        int picture_flip_value = myUtil.getPictureFlip();
+
         int rotation = CameraUtil.getJpegRotation(mCameraId, mOrientation);
         mParameters.setRotation(rotation);
         if (rotation == 90 || rotation == 270) {
diff --git a/src/com/android/camera/util/PersistUtil.java b/src/com/android/camera/util/PersistUtil.java
index c0f4455fd..1a4601806 100644
--- a/src/com/android/camera/util/PersistUtil.java
+++ b/src/com/android/camera/util/PersistUtil.java
@@ -107,11 +107,11 @@ public class PersistUtil {
             SystemProperties.getBoolean("persist.vendor.camera.cs.dumpdepth", false);
     private static final boolean PERSIST_DISABLE_QCOM_MISC_SETTING =
             SystemProperties.getBoolean("persist.vendor.camera.qcom.misc.disable", false);
-    private static final int PREVIEW_FLIP_VALUE =
+    private final int PREVIEW_FLIP_VALUE =
             SystemProperties.getInt("persist.vendor.debug.camera.preview.flip", 0);
-    private static final int PERSIST_VIDEO_FLIP_VALUE =
+    private final int PERSIST_VIDEO_FLIP_VALUE =
             SystemProperties.getInt("persist.vendor.debug.camera.video.flip", 0);
-    private static final int PERSIST_PICTURE_FLIP_VALUE =
+    private final int PERSIST_PICTURE_FLIP_VALUE =
             SystemProperties.getInt("persist.vendor.debug.camera.picture.flip", 0);
     private static final boolean PERSIST_YV_12_FORMAT_ENABLED =
             SystemProperties.getBoolean("persist.vendor.camera.debug.camera.yv12", false);
@@ -268,15 +268,15 @@ public class PersistUtil {
         return PERSIST_DISABLE_QCOM_MISC_SETTING;
     }
 
-    public static int getPreviewFlip() {
+    public int getPreviewFlip() {
         return PREVIEW_FLIP_VALUE;
     }
 
-    public static int getVideoFlip() {
+    public int getVideoFlip() {
         return PERSIST_VIDEO_FLIP_VALUE;
     }
 
-    public static int getPictureFlip() {
+    public int getPictureFlip() {
         return PERSIST_PICTURE_FLIP_VALUE;
     }
 
-- 
2.17.0

